src/main/java/gyro/azure/communication/CommunicationServiceResource.java

```diff
@@ -0,0 +1,282 @@
+     * The Id of the service
+     */
+    @Output
+    @Id
+    public String getId() {
+        return id;
+    }
+
+    public void setId(String id) {
+        this.id = id;
+    }
+
+    @Override
+    public void copyFrom(com.azure.resourcemanager.communication.models.CommunicationServiceResource model) {
+        setResourceGroup(findById(ResourceGroupResource.class, model.resourceGroupName()));
+        setName(model.name());
+        setDataLocation(model.dataLocation());
+        setHostName(model.hostname());
+        setId(model.id());
+
+        getDomains().clear();
+        if (model.linkedDomains() != null) {
+            getDomains().addAll(model.linkedDomains().stream().map(r -> findById(DomainResource.class, r)).collect(
+                Collectors.toList()));
+        }
+
+        getTags().clear();
+        if (model.tags() != null) {
+            getTags().putAll(model.tags());
+        }
+
+        setIdentity(null);
+        if (model.identity() != null) {
+            CommunicationServiceManagedServiceIdentity serviceIdentity =
+                newSubresource(CommunicationServiceManagedServiceIdentity.class);
+            serviceIdentity.copyFrom(model.identity());
+            setIdentity(serviceIdentity);
+        }
+    }
+
+    @Override
+    public boolean refresh() {
+        CommunicationManager client = createClient(CommunicationManager.class);
+
+        com.azure.resourcemanager.communication.models.CommunicationServiceResource service =
+            client.communicationServices().getById(getId());
+
+        if (service == null) {
+            return false;
+        }
+
+        copyFrom(service);
+
+        return true;
+    }
+
+    @Override
+    public void create(GyroUI ui, State state) throws Exception {
+        CommunicationManager client = createClient(CommunicationManager.class);
+
+        CommunicationServiceResourceInner service = new CommunicationServiceResourceInner();
+
+        if (getIdentity() != null) {
+            service.withIdentity(getIdentity().toManagedServiceIdentity());
+        }
+
+        if (!getTags().isEmpty()) {
+            service.withTags(getTags());
+        }
+
+        if (getDataLocation() != null) {
+            service.withDataLocation(getDataLocation());
+        }
+
+        service.withLocation("global");
+        setId(client.serviceClient().getCommunicationServices()
+            .createOrUpdate(getResourceGroup().getName(), getName(), service).id());
+
+        state.save();
+
+        // Add domains in the update call
+        // If domains are not verified but are added to the create call, the api errors out at the service is not saved to the state
+        if (!getDomains().isEmpty()) {
+            service.withLinkedDomains(getDomains().stream().map(DomainResource::getId).collect(Collectors.toList()));
+        }
+
+        client.serviceClient().getCommunicationServices()
+            .createOrUpdate(getResourceGroup().getName(), getName(), service);
```

Comment:
If the domains are added but a wait time error occurs, you should catch the error, verify if all the domains were indeed added. If true, consume the error and return. 
If not decide if you want to wait for them to be added and retry, or save the state with the ones added so that a follow up `gyro up` can create them